client/layout/sidebar/custom-icon.jsx

```diff
@@ -13,20 +13,21 @@
  */
 import React from 'react';
 
-const SidebarCustomIcon = ( { alt, className, icon, ...rest } ) => {
+const SidebarCustomIcon = ( { icon, ...rest } ) => {
 	if ( ! icon ) {
 		return null;
 	}
 
 	if ( icon.indexOf( 'data:image' ) === 0 || icon.indexOf( 'http' ) === 0 ) {
+		const imgStyle = `url("${ icon }")`;
+		const imgStyles = { backgroundImage: imgStyle, maskImage: imgStyle, WebkitMaskImage: imgStyle };
+
 		return (
-			<img
-				alt={ alt || '' }
-				className={ 'sidebar__menu-icon ' + ( className || '' ) }
-				src={ icon }
-				width="20"
+			<span
+				className={ 'sidebar__menu-icon dashicons sidebar__menu-icon-img' }
+				style={ imgStyles }
 				{ ...rest }
```

Comment:
If `alt` is defined in the `rest` object, it will be added to the `span` element, which I think is not completely ok.

Probably, would we like to populate `title` and/or `arial-label` when `alt` is defined?

```es6
const imgData = {};

if ( rest?.alt ) {
	imgData.title = rest.alt;
	imgData.ariaLabel = rest.alt;
	
	// clean image alr property.
	delete rest.alt;
}

// ...

		<span
			className={ 'sidebar__menu-icon dashicons sidebar__menu-icon-img' }
			style={ imgStyles }
			{ ...imgData }
			{ ...rest }
		/>

// ...
```

Reply:
thinking well, maybe we should rename the property to `title` or whatever since it won't be applied to an `<img />` element anymore.

Reply:
I've been looking at this and `alt` is currently never passed as a prop to this component. I think we could consider this a refactor and expect whoever uses it to pass the correct attributes for a `span`. I'm also wondering if it should be `aria-hidden` as it's purely presentational and we'd want screen readers to ignore it.